File: app/src/main/java/net/yanzm/behaviorsubjectsample/CommonData.java
package net.yanzm.behaviorsubjectsample;

import android.support.annotation.NonNull;
import android.support.annotation.Nullable;

/**
 * 各タブで共通のデータ
 */
public class CommonData {

    @NonNull
    public static CommonData empty() {
        return new CommonData(null);
    }

    @Nullable
    private final String data;

    public CommonData(@Nullable String data) {
        this.data = data;
    }

    public boolean isEmpty() {
        return data == null;
    }

    @Nullable
    public String getData() {
        return data;
    }
}

File: app/src/main/java/net/yanzm/behaviorsubjectsample/CommonData.kt
package net.yanzm.behaviorsubjectsample

/**
 * 各タブで共通のデータ
 */
class CommonData(val data: String?) {

    val isEmpty: Boolean
        get() = data == null

    companion object {
        fun empty(): CommonData = CommonData(null)
    }
}

/**
 * タブ固有のデータ
 */
data class SpecificData(val data: List<String>)
